tests/testthat/test-object_usage_linter.R

```diff
@@ -215,3 +215,86 @@ test_that("used symbols are detected correctly", {
     object_usage_linter()
   )
 })
+
+test_that("object_usage_linter finds lints spanning multiple lines", {
+  # Regression test for #507
+  expect_lint(
+    trim_some("
+      foo <- function() {
+        if (unknown_function()) NULL
+
+        if (unknown_function()) {
+          NULL
+        }
+      }
+    "),
+    list(
+      list(message = "unknown_function", line_number = 2L),
+      list(message = "unknown_function", line_number = 4L)
+    ),
+    object_usage_linter()
+  )
+
+  # Linted symbol is not on the first line of the usage warning
+  expect_lint(
+    trim_some("
+      foo <- function(x) {
+        with(
+          x,
+          unknown_symbol
+        )
+      }
+    "),
+    list(message = "unknown_symbol", line_number = 4L, column_number = 5L),
+    object_usage_linter()
+  )
+
+  # Kill regex match to enforce fallback to line 1 column 1 of the warning
+  expect_lint(
+    trim_some("
+      foo <- function(x) {
+        with(
+          x,
+          `\u2019regex_kill`
```

Comment:
the test here is a symbol name that's not matched by our regex? so that the linter has to default to where the expression starts

Reply:
The test causes the `boundary, res$name, boundary` regex to not match against the source code, yes.